libbeat/outputs/elasticsearch/client.go

```diff
@@ -510,18 +512,35 @@ func (conn *Connection) Connect(timeout time.Duration) error {
 	return nil
 }
 
-func (conn *Connection) Ping(timeout time.Duration) (bool, error) {
+// Ping sends a GET request to the Elasticsearch
+func (conn *Connection) Ping(timeout time.Duration) (bool, string, error) {
 	debugf("ES Ping(url=%v, timeout=%v)", conn.URL, timeout)
 
 	conn.http.Timeout = timeout
-	status, _, err := conn.execRequest("HEAD", conn.URL, nil)
+	status, body, err := conn.execRequest("GET", conn.URL, nil)
 	if err != nil {
 		debugf("Ping request failed with: %v", err)
-		return false, err
+		return false, "", err
+	}
+
+	if status >= 300 {
+		return false, "", fmt.Errorf("Non 2xx response code: %d", status)
+	}
+
+	var response struct {
+		Version struct {
+			Number string
+		}
+	}
+
+	err = json.Unmarshal(body, &response)
```

Comment:
make sure body is fully consumed and closed correctly. Otherwise underlying connection can not be re-used (go http module uses keep-alive in order to reuse TCP connections) + need to time out underlying connection in order to close socket.